inspirehep/modules/warnings/ext.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Extension that suppresses warnings.
+
+By default Invenio-Base enables deprecation warnings. This extension
+disables them, unless ``DEBUG_SHOW_DEPRECATION_WARNINGS`` is ``True``.
+
+This extension also disables SQLAlchemy warnings, unless the variable
+``DEBUG_SHOW_SQLALCHEMY_WARNINGS` is set to ``True``.
```

Comment:
>  yet breaking API changes have happened all the time, and will continue to happen because they don't care about us. 

@jacquerie I am a bit hurt your statement. Personally I've invested an enormous amount of time in getting Invenio 3 to where it is today - both at work but also during evening, nights and weekends (take a look at top and bottom line of the contributions on my [GitHub profile](https://github.com/lnielsen) and you might get an idea). My job is not to develop Invenio 3, my job is to run Zenodo, like it's your job to run INSPIRE. However I (perhaps naively) believe that by contributing as much to Invenio 3 as I've done, that I'll get a better product/service in the end. I'm sorry if you feel that we don't care about you, however please try to see it from the other side that it is in fact quite the opposite. I could have decided to just focus on Zenodo, instead I've devoted much more than what is required of me to help build a product that (I hope) is also very useful to you guys and (hopefully) lets you focus on building a great service. 

As for the breaking changes during alpha development of Invenio 3, then I think this is normal and expectable. If we were to provide backward compatibility it would just delay the time at which we could finally declare Invenio 3 fully stable and make a final 3.0 release. And also, the breaking changes was not to bother you, but to get a better product and more clean APIs in the end. Last but not least, I don't feel comfortable declareing any module stable, unless it's been actually battle tested on a production site.
The main goal for Invenio 3 is to have a suite of stable modules with clean APIs and having breaking changes properly managed. This was actually the main reason for initially starting on Invenio 3, and I think it's an absolutely must for running a service on top of Invenio.

So, we do care about you, however we have services to run like you, so I'd like to ask for a bit of patience/understanding until we can finally release Invenio v3.0, or even better help us get there faster for the common good of all our services.